packages/datatrak-web/src/views/AccountSettingsPage/RequestCountryAccessSection/RequestCountryAccessSection.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { ChangeProjectButton } from '../../../components';
+import { AccountSettingsSection } from '../AccountSettingsSection';
+import { AccessGrantedCountryList } from './AccessGrantedCountryList';
+import { RequestCountryAccessForm } from './RequestCountryAccessForm';
+import { useCountryAccessList, useCurrentUser } from '../../../api';
+
+/**
+ * @privateRemarks TODO: Import this from @tupaia/tsutils.
+ *
+ * This is more or less replicated from `typeGuards.ts` in tsutils, because importing it here seems
+ * to indirectly cause DataTrak to crash on remote deployments.
+ *
+ * @see https://beyondessential.slack.com/archives/C032DCHKTGR/p1708553985766539?thread_ts=1707443585.079349&cid=C032DCHKTGR
+ */
+function assertIsNotNullish<T>(val: T): asserts val is NonNullable<T> {
+  if (val === undefined || val === null) {
+    throw new Error(`Expected project to be defined, but got ${val}`);
+  }
+}
```

Comment:
I'm not sure in this use case we want to throw an error quite like this - it's probably not so useful from a user perspective. And given that we shouldn't ever have this situation anyway, since as you mentioned the user would be redirected if they don't have a project selected, I think in the components that refer to the project we can use `?`

Reply:
I hadn’t considered that the error would be user-facing (thought it’d just appear in the console); but it does make sense to file more gracefully than just crashing. Amended!